in_toto/runlib.py

```diff
@@ -235,10 +208,19 @@ def record_artifacts_as_dict(artifacts, exclude_patterns=None,
     # TODO: Do we want to keep the exclude pattern setting?
     exclude_patterns = in_toto.settings.ARTIFACT_EXCLUDE_PATTERNS
 
-  # Apply exclude patterns on the passed artifact paths if available
   if exclude_patterns:
     securesystemslib.formats.NAMES_SCHEMA.check_match(exclude_patterns)
-    norm_artifacts = _apply_exclude_patterns(norm_artifacts, exclude_patterns)
+
+  # Set resolver configs
+  RESOLVER_PARAMS["follow_symlink_dirs"] = follow_symlink_dirs
+  RESOLVER_PARAMS["normalize_line_endings"] = normalize_line_endings
```

Comment:
Am not a big fan of patching globals to configure behavior in methods -- I know, the way I use `in_toto.settings` around here suggests something else, but I learned from past mistakes, see e.g. related secure-systems-lab/securesystemslib#219.


Maybe we can instead configure the resolver directly? E.g.
```python
resolver = Resolver.from_uri(uri)

if isinstance(resolver, FileResolver):
  resolver.follow_symlink_dirs = follow_symlink_dirs
  resolver.normalize_line_endings = normalize_line_endings

resolver.resolve_uri_to_uris(uri) 
```

Reply:
Speaking of FileResolver specific behavior, can't we also hide the base_path case handling inside the file resolver? It doesn't seem very useful to `chdir` if the URIs are not file paths.

Reply:
I'd argue that, if you need to chdir to resolve artifacts, then you are probably using a sort of file resolver, even if you are looking for very specific files like git objects.